addons/mail/static/tests/discuss/core/public_web/sub_channels.test.js

```diff
@@ -192,3 +193,75 @@ test("mention suggestions in thread match channel restrictions", async () => {
     await contains(".o-mail-Composer-suggestion", { text: "Mitchell Admin" });
     await contains(".o-mail-Composer-suggestion", { text: "p1" });
 });
+
+test("sub channel is muted (opacity-50) when mute is active", async () => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "General" });
+    await start();
+    await openDiscuss(channelId);
+    await contains(".o-mail-Discuss-threadName", { value: "General" });
+    await click("button[title='Threads']");
+    await click("button[aria-label='Create Thread']");
+    await contains(".o-mail-Discuss-threadName", { value: "New Thread" });
+    await contains(".o-mail-DiscussSidebarChannel-subChannel.opacity-50", { count: 0 });
+    await click("button[title='Notification Settings']");
+    await click("button:contains('Mute Conversation')");
+    await click("button:contains('Until I turn it back on')");
+    await contains(".o-mail-DiscussSidebarChannel-subChannel.opacity-50", { count: 1 });
+});
+
+test("sub channel is not shown when parent channel is muted", async () => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "General" });
+    await start();
+    await openDiscuss(channelId);
+    await contains(".o-mail-Discuss-threadName", { value: "General" });
+    await click("button[title='Threads']");
+    await click("button[aria-label='Create Thread']");
+    await contains(".o-mail-Discuss-threadName", { value: "New Thread" });
+    await contains(".o-mail-DiscussSidebarChannel-subChannel", { count: 1 });
+    await click(".o-mail-DiscussSidebarChannel");
+    await click("button[title='Notification Settings']");
+    await click("button:contains('Mute Conversation')");
+    await click("button:contains('Until I turn it back on')");
+    await contains(".o-mail-DiscussSidebarChannel-subChannel", { count: 0 });
+});
+
+test("sub channel is shown when parent channel is muted but sub channel has unread messages", async () => {
```

Comment:
"sub-channel" terminology is poor, prefer "channel thread" or "sub-thread" (which is the name we expose functionally). Using functionally relevant names is preferred, because the specs match functional names. Technical names should try their best to match functional names, but sometimes it costs too much to change them unfortunately :(